### PR TITLE
[BugFix] [Infrastructure] Verify if CCE IDs used in OVAL and XCCDF have required 'CCE-XXXX-X' or 'CCE-XXXXX-X'. Drop those not meeting the requirement

### PR DESCRIPTION
**Pull request by iankko**
The ScapVal (v1.2.14.1) specifies the following form of the resulting CCE identifier:
```
CCE number is in an invalid format or the check-digit does not match. It should
be of format CCE-XXXX-X or CCE-XXXXX-X where each X is a digit, and the
final X is a check-digit. 
```
from http://people.redhat.com/swells/nist-scap-validation/scap-val-requirements-1.2.html ```Requirement A17```.

Though in current ```RHEL/6``` master content all CCE IDs are in valid form (they have been already assigned from the pool), for the  ```RHEL/7``` product (and possibly also others) the CCE IDs have "intermediary / temporary" form of ```CCE-RHEL7-TBD``` (or alternative of this one).

This is clearly not meeting the requirement, causing resulting XCCDF / DataStream benchmark to fail the ScapVal validation.

Therefore:
* when propagating CCE ID information from XCCDF to OVAL verify the corresponding CCE ID to be propagated is in valid / required form (skip the propagation if the form is invalid) - patch https://github.com/OpenSCAP/scap-security-guide/commit/f5dd072b3886ca645911f7f74989830448172236 is performing this,
* when producing final XCCDF (via ```relabelids.py``` helper script) scan the output XCCDF to verify if all referenced CCE IDs have one of the above required two forms. In the case referenced CCE ID in invalid form is found, drop such CCE ID (e.g. ```CCE-RHEL7-TBD``` reference) from the final XCCDF (and therefore subsequently also from DataStream output) - patch https://github.com/OpenSCAP/scap-security-guide/commit/913ce8013a29c42ea4ab6cb71256e32843fff29f is ensuring this behaviour

Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1230
Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1229
Fixes: https://github.com/OpenSCAP/scap-security-guide/issues/1228

Please review.

Thank you, Jan.

### CODE REVIEW

**Comment by redhatrises** [comment]
I like this idea better than commenting out the `ident` tags. I will review this.

**Comment by redhatrises** [comment]
LGTM. Ack.

**Comment by iankko** [comment]
@redhatrises 
> I like this idea better than commenting out the ident tags. I will review this.

Thanks for the review, Gabe. Appreciated!

Yeah, I have considered the option just to comment out the invalid CCE IDs for now. But that approach would require:
* one huge patch for the beginning (to comment out all of the invalid ones in RHEL/7 and also other benchmarks),
* continuous monitoring / synchronization, if the invalid ones are still commented out (IOW periodically to check if ```scap-security-guide-nist-testsuite``` Jenkins CI testing job is still passing).

In short, IMHO that approach wouldn't scale much. Therefore went different way (let the build system itself to pass in the valid ones [having correct form] and filter out the invalid ones). This way  as soon as valid CCE is assigned for new e.g. RHEL/7 rule, it will be automatically included into the final benchmark (without the need we periodically to check if ```scap-security-guide-nist-testsuite``` Jenkins CI testing job is continuing to pass).
